### PR TITLE
[DOCS] Reinforce connection between rollover and index age

### LINKED ISSUE #85651: [DOCS] Better documentation around ILM rollover and min_age
Consider an example ILM policy like the following:

```
{
  "phases": {
    "hot": {
      "actions": {
        "rollover": {
          "max_age": "7d"
        }
      }
    },
    "warm": {
      "min_age": "14d",
      "actions": {
        "readonly": {}
      }
    },
    "delete": {
      "min_age": "28d",
      "actions": {
        "delete": {}
      }
    }
  }
}
```

I think it would be tempting to look at this policy and expect that an index created on 2022-04-01 with that policy would move to the warm phase on 2022-04-15 and finally be deleted on 2022-04-29. However, in an ILM policy with a `rollover` action, the `rollover` will reset the index's `"age"` to 0 (as reported by `_ilm/explain`), see https://github.com/elastic/elasticsearch/pull/30853.

Since the rollover resets the `"age"` to 0, a nice way of thinking about this is that **the phase transition times post-rollover are relative to when the index rolled over** (_not to when the index was originally created_).

Note, however, that in a policy without a `rollover`, the index `"age"` would not have been reset, and consequently the phase transition times would be relative to when the index was created (one caveat: unless the `index.lifecycle.origination_date` setting were used for that index, see the [ILM settings documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/ilm-settings.html#_index_level_settings)).

-----

#84273 touched on this, specifically, it added a note onto https://www.elastic.co/guide/en/elasticsearch/reference/current/getting-started-index-lifecycle-management.html (ctrl+f for `relative to`).

And there's also a note on https://www.elastic.co/guide/en/elasticsearch/reference/current/ilm-rollover.html itself that goes along with the last example (https://www.elastic.co/guide/en/elasticsearch/reference/current/ilm-rollover.html#_rollover_condition_blocks_phase_transition), but even knowing how things work, I still find that explanation to be a bit on the subtle side.

-----

Some places we might consider giving this top billing or at least more attention:

- https://www.elastic.co/guide/en/elasticsearch/reference/current/ilm-index-lifecycle.html the high level 'conceptual' view of the index lifecycle, it mentions transitions and index ages, but not this behavior.
- https://www.elastic.co/guide/en/elasticsearch/reference/current/index-rollover.html the high level rollover documentation
- https://www.elastic.co/guide/en/elasticsearch/reference/current/ilm-rollover.html the lower level rollover action documentation (n.b. to be fair, as previously stated, it is mentioned as sort of an aside in [the last example](https://www.elastic.co/guide/en/elasticsearch/reference/current/ilm-rollover.html#_rollover_condition_blocks_phase_transition) on that page).

I think we could add a note to each of those pages, perhaps at the risk of being a little repetitive. However, given that this comes up fairly frequently as a point of confusion for our users, it might even make sense to go above and beyond that to provide more of a long form explainer, too, like a "here's how this works, and *why* it works that way" so that we give people a very good explanation on this if it happens to slip by them and then later trip them up.

### PR DESCRIPTION
**Pull request by shainaraskas**
- Fixes https://github.com/elastic/elasticsearch/issues/85651
- Adds notes to rollover + ilm tutorials that explains how the index creation date and therefore its `min_age` is impacted by rollover
- Links to [How `min_age` is calculated](https://www.elastic.co/guide/en/elasticsearch/reference/master/index-lifecycle-error-handling.html#_how_min_age_is_calculated) as the source of truth

### CODE REVIEW

File: docs/reference/ilm/actions/ilm-rollover.asciidoc [hunk 1]
```diff
@@ -7,6 +7,13 @@ Phases allowed: hot.
 Rolls over a target to a new index when the existing index satisfies
 the specified rollover conditions.
 
+[NOTE]
+====
+When an index is rolled over, a new index is created with a `creation_date` that reflects
+the rollover time. This impacts the calculation of downstream {ilm} `max_age` and `min_age` conditions. 
+<<min-age-calculation,Learn more>>.
```
Comment (dakrone) [inline docs/reference/ilm/actions/ilm-rollover.asciidoc:14]: I think the "This" in the second sentence is ambiguous. For instance, after an index is rolled over, there is no "downstream ILM `max_age`" condition, the only time comparison is for the `min_age` of phase definitions.

I think we could clarify this to something like:
```suggestion
When an index is rolled over, a new index is created with a new `creation_date` while the previous index
is updated with a timestamp that reflects the rollover time. This rollover time impacts the calculation of downstream {ilm} `min_age` phase calculations. 
<<min-age-calculation,Learn more>>.
```

(feel free to change this, I just put it here as a suggestion)

**Review by dakrone — CHANGES_REQUESTED** [review]

File: docs/reference/ilm/actions/ilm-rollover.asciidoc (hunk 1 above)
Comment (dakrone) [reply docs/reference/ilm/actions/ilm-rollover.asciidoc:14]: > isn't it the `creation_date` that's used in `min_age` phase calculation?

No, ILM uses the rollover time (the time at which the index was rolled over). In your suggestion it makes it sound as though the new index has a `creation_date` which then gets used for ILM `min_age` phase calculations, when it's really the _old_ write index that has a new "time" set (the rollover time) that is used for the `min_age` calculations. The newly created index is back at the beginning again, so it's likely back to the `rollover` step.

For example, if a user has a policy where the `hot` phase is configured with a rollover action for `max_docs: 1`, and a `warm` phase with a `min_age: 1h`. Say it takes 10 minutes before a single document is indexed (to meet the `max_docs: 1` condition), so it rolls over at the 10 minute mark. The index records the time it rolls over, then waits an additional 30 minutes from this time (for a total of 40 minutes) before entering the `warm` phase.

**Comment by shainaraskas** [comment]
hey @dakrone, I've taken another shot at this. Looks like the original explanation in the docs might have been wrong so I made some additional updates. thanks for your explanation and your patience!

File: docs/reference/ilm/index-rollover.asciidoc [hunk 2]
```diff
@@ -35,8 +34,15 @@ more configuration steps and concepts:
...
 * An _index alias_ that references the entire set of indices. 
 * A single index designated as the _write index_.
-This is the active index that handles all write requests.
-On each rollover, the new index becomes the write index.
+This is the active index that handles all write requests. 
+On each rollover, the new index becomes the write index. 
+
+[NOTE]
+====
+When an index is rolled over, previous index's age is updated to reflect the rollover time.
```
Comment (dakrone) [inline docs/reference/ilm/index-rollover.asciidoc:42]:
```suggestion
When an index is rolled over, the previous index's age is updated to reflect the rollover time. 
```

**Review by dakrone — APPROVED** [review]
LGTM, thanks for clarifying this! I left two comments about typos that I saw but otherwise it looks great.

File: docs/reference/ilm/actions/ilm-rollover.asciidoc [hunk 3]
```diff
@@ -7,6 +7,13 @@ Phases allowed: hot.
 Rolls over a target to a new index when the existing index satisfies
 the specified rollover conditions.
 
+[NOTE]
+====
+When an index is rolled over, previous index's age is updated to reflect the rollover time.
```
[the remark this answers is not in the document]
Comment (shainaraskas) [reply docs/reference/ilm/actions/ilm-rollover.asciidoc:12]: great catch! ty

**Comment by shainaraskas** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.14|[](https://github.com/elastic/elasticsearch/pull/109693)|
|✅|8.13|[](https://github.com/elastic/elasticsearch/pull/109694)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

**Pull request merged**